pkg/internal/codegen/parse/util.go

```diff
@@ -311,6 +325,16 @@ func getCategoriesTag(c *types.Type) string {
 	return resource
 }
 
+// getSingularName returns the value of the +kubebuilder:singular tag
+func getSingularName(c *types.Type) string {
+	comments := Comments(c.CommentLines)
+	singular := comments.getTag("kubebuilder:singular", "=")
+	if len(singular) == 0 {
+		panic(errors.Errorf("Must specify +kubebuilder:singular comment for type %v", c.Name))
```

Comment:
this error doesn't make any sense with how singular is used.  maybe reword to say "must specify a value to use with singular" or something?

Reply:
@damemi  could you please add one test case here for this change? You can add your test case here https://github.com/kubernetes-sigs/controller-tools/blob/master/pkg/crd/generator/testData/pkg/apis/fun/v1alpha1/toy_types.go#L97
and run this file to see if you're getting the proper o/p.
https://github.com/kubernetes-sigs/controller-tools/blob/master/pkg/crd/generator/generator_test.go